src/supla/sensor/HX711.h

```diff
@@ -0,0 +1,161 @@
+*/
+
+#ifndef SRC_SUPLA_SENSOR_HX711_H_
+#define SRC_SUPLA_SENSOR_HX711_H_
+
+/*
+Dependency: https://github.com/olkal/HX711_ADC,
+use library manager to install it
+*/
+
+#include <HX711_ADC.h>
+
+#include "weight.h"
+#include "../storage/storage.h"
+
+namespace Supla {
+namespace Sensor {
+
+#pragma pack(push, 1)
+struct HX711ConfigData {
+  uint32_t tareOffset;
+  float calValue;
+};
+#pragma pack(pop)
+
+class HX711 : public Weight {
+ public:
+  HX711(int sdaPin, int sclPin, float defCalValue = 696.0)
+      : sdaPin(sdaPin), sclPin(sclPin), defCalValue(defCalValue),
+          hx711Sensor(sdaPin, sclPin) {
+  }
+
+  double getValue() {
+    if ((hx711Sensor.getSPS() < 7) || (hx711Sensor.getSPS() > 100)) {
+      value = WEIGHT_NOT_AVAILABLE;
+      SUPLA_LOG_DEBUG("HX711 sampling rate is abnormal, check connection");
+    } else {
+      value = (value < 0 && value > -1) ? 0 : value;
+    }
+    return value;
+  }
+
+  void onInit() {
+    hx711Sensor.begin();
+    hx711Sensor.setTareOffset(tareOffset);
+    uint64_t stabilizingTime = 2000;
+    bool _tare = false;
+    hx711Sensor.start(stabilizingTime, _tare);
+    if (hx711Sensor.getTareTimeoutFlag()) {
+      SUPLA_LOG_DEBUG("HX711 timeout, check connection");
+    } else {
+      calFactor = (calValue > 0) ? calValue : defCalValue;
+      hx711Sensor.setCalFactor(calFactor);
+      SUPLA_LOG_DEBUG("HX711 startup is complete");
+    }
+    while (!hx711Sensor.update()) {}
+    SUPLA_LOG_INFO("HX711 calibration value: %.2f", hx711Sensor.getCalFactor());
+    SUPLA_LOG_INFO(
+        "HX711 measured conversion time ms: %.2f",
+         hx711Sensor.getConversionTime());
+    SUPLA_LOG_DEBUG(
+        "HX711 measured sampling rate HZ: %.2f", hx711Sensor.getSPS());
+    SUPLA_LOG_INFO(
+        "HX711 measured settlingtime ms: %d", hx711Sensor.getSettlingTime());
+    if ((hx711Sensor.getSPS() < 7) || (hx711Sensor.getSPS() > 100)) {
+      SUPLA_LOG_DEBUG("HX711 sampling rate is abnormal, check connection");
+    }
+    channel.setNewValue(getValue());
+  }
+
+  void calibrateScales(float knownMass = 0) {
+    if (!readyToCalibration) {
+      SUPLA_LOG_DEBUG("HX711 please do tare");
+    } else {
+      hx711Sensor.update();
+      hx711Sensor.refreshDataSet();
+      if (knownMass > 0) {
+        calValue = hx711Sensor.getNewCalibration(knownMass);
+        SUPLA_LOG_DEBUG("HX711 new calibration value: %.2f", calValue);
+        Supla::Storage::ScheduleSave(100);
+      } else {
+        SUPLA_LOG_DEBUG("HX711 known mass must be greater than zero");
+      }
+    }
+    readyToCalibration = false;
+  }
+
+  void tareScales() {
+    hx711Sensor.tare();
+    tareOffset = hx711Sensor.getTareOffset();
+    SUPLA_LOG_DEBUG("HX711 new tare offset: %d", tareOffset);
+    Supla::Storage::ScheduleSave(100);
+    readyToCalibration = true;
+  }
+
+  void iterateAlways() {
+    if (hx711Sensor.update()) {
+      value = hx711Sensor.getData();
+    }
```

Comment:
You can move  this code to below "if". Then it will call update and getData only every 500 ms, when it is needed.

Reply:
According to the library, the sampling frequency must be in the range 7 - 100 Hz. Now, it's around 10.5 Hz.